### ISSUE TITLE
Importing Compass + Singularitygs's background-grid mixin causes SassMeister being unable to compile the stylesheet 

### ISSUE THREAD

**Issue by lozandier** [at-import/Singularity#168]
Attempting to import Compass (`@import "compass"`) and then trying to use singularitygs's `background-grid` causes SassMeister to fail in its attempt to compile the stylesheet. 

```
wrong number of arguments (25 for 1..2) for `_linear-gradient_legacy'
```

Version of Compass being used by Singularity: `1.0.0.alpha.18`
Version of Sass: `3.3.0`
Version of Singularitygs: `1.2.0.rc.6`

http://sassmeister.com/gist/9478983

**Comment by lozandier** [comment]
@lolmaus It's not related to any sassmeister gist, including a blank one,  but I'll create one in a few moments showing this behavior. 

## Steps to Reproduce The Problem
1. @import compass 
2. @import singularity 
3. `+sgs-change("debug", true)` or `@include sgs-change("debug", true) 
4. Use `+background-grid` or `@include background-grid`
5. Error occurs.

## Example

An example from a gist I happen to have open. 

### Before

![screen shot 2014-03-11 at 4 36 57 am](https://f.cloud.github.com/assets/1223224/2384757/9cbeb774-a912-11e3-9fd7-546795b088fc.png)

### After

![screen shot 2014-03-11 at 4 38 57 am](https://f.cloud.github.com/assets/1223224/2384761/a8eccc2a-a912-11e3-9753-cfa9b9a632a2.png)

**Comment by lozandier** [comment]
@lolmaus Gist provided; can be found [here](http://sassmeister.com/gist/9484198)

**Comment by Snugug** [comment]
No need to keep going, I know what the issue is. Working to fix

**Comment by lozandier** [comment]
The issue still happens. I just  tried to use Singularity 1.2 & Compass 1.0.0.alpha18 and still got the same error.

**Comment by primozs** [comment]
Same issue here Singularity 1.2 & Compass 1.0.0.alpha19
It seams like linear-gradient function problem

**Comment by kogakure** [comment]
I ran into the same issue. Without Compass it’s working (Demo: http://sassmeister.com/gist/9992763). When using Compass the error occurs. Using Compass 1.0.0.alpha19.

**Comment by Snugug** [comment]
It looks like the `linear-gradient` function in Compass overrides the standard `linear-gradient` that we use, and [only works when used inside of a Compass mixin](http://beta.compass-style.org/reference/compass/css3/images/#function-linear-gradient). Therefore, I think we're going to need to interpolate what we're outputting in some way.

**Comment by Snugug** [comment]
[Relevant Code](https://github.com/Team-Sass/Singularity/blob/1.x.x/stylesheets/singularitygs/helpers/_background-grid.scss#L220-L224)

Suggested Fix:

`background-image: unquote('linear-gradient(to #{opposite-direction($Direction)}, #{$Gradient})');`

**Comment by primozs** [comment]
in my case then there is another problem in background-build mixin:

```
Syntax error: Incompatible units: 'em' and '%'.
        on line 201 of /Users/primozsusa/.rvm/gems/ruby-1.9.3-p286/gems/singularitygs-1.2.0/stylesheets/singularitygs/helpers/_background-grid.scss, in `background-build'
        from line 52 of /Users/primozsusa/.rvm/gems/ruby-1.9.3-p286/gems/singularitygs-1.2.0/stylesheets/singularitygs/helpers/_background-grid.scss, in `background-grid'
```

**Comment by grayghostvisuals** [comment]
@Snugug 

**sass file**

``` sass
@include sgs-change('debug', true);

.grid {
    @include background-grid;
}
```

**Suggested Fix above: `singularitygs/helpers/_background-grid.scss`**

``` sass
@if $bkpt != -1px {
      @include breakpoint($bkpt) {
        // background-image: linear-gradient(to opposite-direction($Direction), $Gradient);
        background-image: unquote('linear-gradient(to #{opposite-direction($Direction)}, #{$Gradient})');
      }
    }
    @else {
      // background-image: linear-gradient(to opposite-direction($Direction), $Gradient);
      background-image: unquote('linear-gradient(to #{opposite-direction($Direction)}, #{$Gradient})');
    }
```

**Result**

``` bash
Running "compass:dist" (compass) task
    error css/src/styles.scss (Line 172 of _background-grid.scss: Function column-count finished without @return)
overwrite css/styles.css
Compilation took 1.412s
```

**Comment by Snugug** [comment]
The fix isn't implemented yet, however, initial testing suggests you should not be getting those results. Those were really notes for me.

**Comment by niairo** [comment]
Just to say, that the Suggested Fix worked well for me : 
Using compass-core 1.0.0.alpha.19
Using singularitygs 1.2.0

**Comment by jottr** [comment]
@Snugug do you have any suggestions what we do until a fix can be released?
I am beginning a new project and would like to use singularity grids in conjunction with compass.
Since 1.2.0 introduced a new syntax, I don't see the point in downgrading to a previous release.

**Comment by Snugug** [comment]
Just released v1.2.1 that includes my recommended fix

**Comment by jottr** [comment]
Perfect! Thank you for fixing this @Snugug
